frontend-react/src/hooks/UseValueSets.ts

```diff
@@ -1,76 +1,84 @@
 
 export interface TableAttributes {
     version: number;
     createdAt?: string;
     createdBy?: string;
 }
 
-/*
-
-  Fetch / Mutate functions and helpersa
-
-*/
-
-/*
-TO DISCUSS: we could set the pattern to be something like this or
-we could reference query keys within the function https://stackoverflow.com/a/68111112/5500298
-
-Have not tested, but I feel like to get the full value of react-query's refetching behavior we may want to go with the last option.
-When react-query refetches it will just re-run the function we passed to it, this would not take into account changes in variable dependencies
-unless we bake those into function itself?
-
-Note that I ran into type errors when trying to use higher order functions to avoid the inlining
-*/
-
-// will return either a value set or value set row depeding on
-// value set table name
-// we may not need the generic here, react-query probably handles that. But it couldn't hurt?
-const getLatestData = <T>(tableName: string, version: number) => {
-    const endpointHeader = lookupTableApi.getTableData<T>(version, tableName);
-
-    return axios
-        .get<T>(endpointHeader.url, endpointHeader)
-        .then((response) => response.data);
+// these can be stored on our resource once we build that out
+// notice the react-router style colon demarcated dynamic path segments
+const getTableListConfig = {
+    path: "/lookuptables/list",
+    method: HTTPMethods.GET,
+};
+const getTableDataConfig = {
+    path: "/lookuptables/:tableName/:version/content",
+    method: HTTPMethods.GET,
 };
 
-// not going to type this yet, as this function will likely go away when we bring in fetch by provider
-// returns a list of all lookup tables to be filtered
-const getLookupTables = () =>
-    axios(lookupTableApi.getTableList()).then(({ data }) => data);
+const updateTableConfig = {
+    path: "/lookuptables/:tableName",
+    method: HTTPMethods.POST,
+};
+const activateTableConfig = {
+    path: "/lookuptables/:tableName/:version/activate",
+    method: HTTPMethods.PUT,
+};
 
-const endpointHeaderUpdate = lookupTableApi.saveTableData(
-    LookupTables.VALUE_SET_ROW
-);
+interface Segments {
+    [segmentKey: string]: string;
+}
 
-const updateValueSet = (data: ValueSetRow[]) =>
-    axios
-        .post(endpointHeaderUpdate.url, data, endpointHeaderUpdate)
-        .then(({ data }) => data);
+interface AxiosOptionsWithSegments extends AxiosRequestConfig {
+    segments: Segments;
+}
 
-const activateValueSet = (tableVersion: number) => {
-    const endpointHeaderActivate = lookupTableApi.activateTableData(
-        tableVersion,
-        LookupTables.VALUE_SET_ROW
+// takes a map of path segment keys (as defined by colons in config paths declarations, such as in react outer)
+// to segment values for a particular API call
+// in order to produce concrete path from a dynamic one
+// this stuff likely will live on even in a world with resources, one way or another
```

Comment:
Yep, I still need to think through a few details, but part of the keys to how I'm trying to structure this to, except in maybe some very, very simple cases that I can't conceive of yet, have components communicate with the API only through custom hooks that we can type and implement, to some degree, however we want or need to. The custom hooks would then be the ones leveraging this authorized fetch system and react-query, but we can control that interaction. So, in the case of a delete mutation hook, we may have the custom hook take a particular resource, if that helps us in terms of managing endpoint configuration, but in the implementation of the hook, we can specify that we expect a different resource (or no resource) back as a return value. It may take a few more steps to customize this way, and may look a little hacky - since so far I am expecting there to be a `data` attribute returned on all successful axios calls - but it won't be impossible.